pkg/controllers/resourceapply/config_hash_test.go

```diff
@@ -0,0 +1,271 @@
+	return []corev1.Container{
+		{
+			Name: "foo", EnvFrom: []corev1.EnvFromSource{
+				{
+					ConfigMapRef: &corev1.ConfigMapEnvSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: "envConfigMap"},
+					},
+				},
+			},
+		},
+		{
+			Name: "bar", EnvFrom: []corev1.EnvFromSource{
+				{
+					SecretRef: &corev1.SecretEnvSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: "envSecret"},
+					},
+				},
+			},
+		},
+		{
+			Name: "fizz", Env: []corev1.EnvVar{
+				{
+					ValueFrom: &corev1.EnvVarSource{
+						ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{Name: "envConfigMap2"},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "fizz",
+			Env: []corev1.EnvVar{
+				{
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{Name: "envSecret2"},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestCollectDependantConfigs(t *testing.T) {
+	tcs := []struct {
+		name               string
+		podTemplate        *corev1.PodTemplateSpec
+		expectedSecrets    []string
+		expectedConfigMaps []string
+	}{
+		{
+			name:               "nil pod template spec",
+			podTemplate:        nil,
+			expectedSecrets:    []string{},
+			expectedConfigMaps: []string{},
+		},
+		{
+			name: "Empty pod template spec",
+			podTemplate: &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{},
+			},
+			expectedSecrets:    []string{},
+			expectedConfigMaps: []string{},
+		},
+		{
+			name: "secret and config map volumes",
+			podTemplate: &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Volumes: getVolumes(),
+				},
+			},
+			expectedSecrets:    []string{"secret", "secret2"},
+			expectedConfigMaps: []string{"configmap", "configmap2"},
+		},
+		{
+			name: "container secrets and configmaps",
+			podTemplate: &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: getContainers(),
+				},
+			},
+			expectedSecrets:    []string{"envSecret", "envSecret2"},
+			expectedConfigMaps: []string{"envConfigMap", "envConfigMap2"},
+		},
+		{
+			name: "initContainer secrets and configmaps",
+			podTemplate: &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					InitContainers: getContainers(),
+				},
+			},
+			expectedSecrets:    []string{"envSecret", "envSecret2"},
+			expectedConfigMaps: []string{"envConfigMap", "envConfigMap2"},
+		},
+		{
+			name: "everything",
+			podTemplate: &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers:     getContainers()[:2],
+					InitContainers: getContainers()[2:],
+					Volumes:        getVolumes(),
+				},
+			},
+			expectedSecrets:    []string{"envSecret", "envSecret2", "secret", "secret2"},
+			expectedConfigMaps: []string{"configmap", "configmap2", "envConfigMap", "envConfigMap2"},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			g := gmg.NewWithT(t)
```

Comment:
i haven't used this function before, what does it do here?

Reply:
Intializes gomega matcher for using things like `Expect`.

https://pkg.go.dev/github.com/onsi/gomega#NewWithT